streaming/src/main/scala/org/apache/spark/streaming/dstream/FileInputDStream.scala

```diff
@@ -163,8 +166,11 @@ class FileInputDStream[K, V, F <: NewInputFormat[K, V]](
   /** Clear the old time-to-files mappings along with old RDDs */
   protected[streaming] override def clearMetadata(time: Time) {
     super.clearMetadata(time)
-    val oldFiles = batchTimeToSelectedFiles.filter(_._1 < (time - rememberDuration))
-    batchTimeToSelectedFiles --= oldFiles.keys
+    val oldFiles = batchTimeToSelectedFiles.synchronized
+    {
+      batchTimeToSelectedFiles.filter(_._1 < (time - rememberDuration))
```

Comment:
@srowen 
Thank you very much for your comments.  I changed the code to synchronize the whole block. 
 The extra parentheses at line 153 seems to be required.  Removing it causes problem.